src/ripple/ledger/impl/ApplyStateTable.cpp

```diff
@@ -642,8 +647,8 @@ ApplyStateTable::threadOwners (ReadView const& base,
     default:
     {
         // If sfAccount is present, thread to that account
-        if ((*sle)[~sfAccount])
-            threadTx (base, meta, (*sle)[sfAccount], mods, j);
+        if (auto const optSleAcct {(*sle)[~sfAccount]})
```

Comment:
@HowardHinnant, darn it.  Now I have to rethink my position.  I wasn't paying close enough attention during Nico's presentation.  Here in his presentation (https://youtu.be/7DTlWPgX6zs?t=2554) he talks about how braced initialization without the `=` ignores the `explicit` on explicit ctors.  Yuck.

Back to the drawing board.  And I thought Nico had solved initialization for me.  Maybe the right answer for me will e always using braced initialization _with_ the equals sign.  I'm not sure now.

It would be great if you and Nico had a chat at some point in the future.  I'd really like to understand why Nico thinks it is desirable to ignore whether the ctor is `explicit` or not.

Reply:
I'm still figuring this out too.  I strongly suspect that if and when I do figure it out, it won't be to use the same style all the time.

Reply:
It's beginning to smell that way to me too.  It feels like `= {}` is the safest form of initialization.  Then, depending on the form of unsafeness you want you either...

* Lose the `=` (which uses explicit constructors), or
* Replace the `{}` with `()` which allows narrowing.

So the person reading the code can see the intended conversions based on the form of initialization.

If that's correct then C++ is turning into an even more subtle language, at least when it's being used cautiously.  The less cautious form is to take Nico's advice.  But I don't think I want to be that cavalier.